File: src/main/Java/boj_step10/boj1085.java
import java.io.*;
import java.util.StringTokenizer;

public class boj1085 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int x = Integer.parseInt(tokenizer.nextToken());
        int y = Integer.parseInt(tokenizer.nextToken());
        int w = Integer.parseInt(tokenizer.nextToken());
        int h = Integer.parseInt(tokenizer.nextToken());
        if (x <= w / 2) w = 0;
        if (y <= h / 2) h = 0;
        int distanceX = Math.abs(w - x);
        int distanceY = Math.abs(h - y);
        int distance = Math.min(distanceX, distanceY);
        writer.append(String.valueOf(distance)).close();
    }
}

File: src/main/boj_step10/boj1085.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() = with(BufferedWriter(OutputStreamWriter(System.out))) {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val (x, y, w, h) = r.readLine().split(" ").map { it.toInt() }
    var min = kotlin.math.min(x, y)
    min = kotlin.math.min(min, w - x)
    min = kotlin.math.min(min, h - y)
    write(min.toString())
    close()
}